server/src/main/java/org/elasticsearch/index/mapper/ParseContext.java

```diff
@@ -596,6 +596,12 @@ public ContentPath path() {
         };
     }
 
+    /**
+     * We are actively deprecating and removing the ability to pass complex objects
```

Comment:
Small comment, I think we usually style it like `@deprecated <explanation text>`